system/lib/wasmfs/syscalls.cpp

```diff
@@ -20,11 +20,50 @@
 #include <utility>
 #include <vector>
 #include <wasi/api.h>
+#include <unistd.h>
 
 extern "C" {
 
 using namespace wasmfs;
 
+// Copy the file specified by the pathname into JS.
+// Return a pointer to the JS buffer in HEAPU8.
+void* emscripten_wasmfs_read_file(char* path) {
+  struct stat file;
+  int err = 0;
+  err = stat(path, &file);
+  if (err < 0) {
+    emscripten_console_error(
+      "Fatal error in FS.readFile");
+    abort();
+  }
+  
+  size_t size = file.st_size;
```

Comment:
Maybe we should use `uint32_t size` to ensure it will always be 4 bytes even in wasm64 mode. It's probably worth checking that the size fits in 32 bits when we do that.